packages/rpc-transport/src/transports/http/http-transport.ts

```diff
@@ -10,10 +10,13 @@ import {
 
 type Config = Readonly<{
     headers?: AllowedHttpRequestHeaders;
-    url: string;
+    url: ClusterUrl;
 }>;
 
-export function createHttpTransport({ headers, url }: Config): IRpcTransport {
+export function createHttpTransport<TConfig extends Config>({
```

Comment:
Introduced a generic arg here since we never had one before and it can be inferred.

Reply:
We could also make `Config` generic.

```ts
type Config<TClusterUrl extends ClusterUrl> = Readonly<{
    headers?: AllowedHttpRequestHeaders;
    url: TClusterUrl;
}>;

export function createHttpTransport<TClusterUrl extends ClusterUrl>({
    headers, 
    url 
}: Config<TClusterUrl>): IRpcTransportFromClusterUrl<TClusterUrl> {...}
```

Reply:
Hah, this was a great call. Slick!